File: OOP_Ponavljanje/c++/dodavanjeProfesoraNasljedivanje.cpp
    #include<iostream>
    using namespace std;

class Osoba {
public:
    string ime;
    Osoba() {}
    Osoba(string ime) {
        this->ime = ime;
    }
    
    string dohvatiIme() {
        return this->ime;
    }
};    
    
    
class Profesor : public Osoba {
public:
    string kolegiji[5];

    Profesor(string ime, string kolegiji[]) {
        this->ime = ime;
        int iterator = 0;
        while(!kolegiji[iterator].empty()) {
            this->kolegiji[iterator] = kolegiji[iterator];
            iterator++;
        }

    }



    void ispisKolegija() {
        int iterator = 0;
        while(!this->kolegiji[iterator].empty()) {
            cout << "Predaje: " << this->kolegiji[iterator] << endl;
            iterator++;
        }
    }

};

class Student : public Osoba {
    int matBr;
    static int numObj;
public:
    Student(string ime, int matBr) {
        Student::numObj++;
        this->ime = ime;
        this->matBr = matBr;
    }

    static int getNumObj() {
        return Student::numObj;
    }

};

int Student::numObj = 0;

int main(int argc, char** argv) {
    Student* pero = new Student("Petar", 1234);
    Student* mate = new Student("Mate", 3432);
    Profesor* profa = new Profesor("Miljenko", new string[5]{"PIN","DIMT","RIP"});
    profa->ispisKolegija();
    cout << profa->dohvatiIme();
    return 0;
}

File: OOP_Ponavljanje/php/dodavanjeProfesoraNasljedivanje.php
<?php 

class Osoba {
	public $ime;

	public function __construct($ime) {
		$this->ime = $ime;
		echo "Base class constructor called!";
	}
}

class Student extends Osoba {
	private $matBr;

	public function __construct($ime, $matBr) {
		$this->ime = $ime;
		$this->matBr = $matBr;
		echo "Created student! <br>".$this->ime."<br>".$this->matBr."<br>";
	}
}

class Profesor extends Osoba {
	public $kolegiji = array();

	public function __construct($ime,array $kolegiji) {
		$this->ime = $ime;
		$this->kolegiji = $kolegiji;
	}

	public function ispisKolegija() {
		foreach ($this->kolegiji as $kolegij)
			echo "<br>".$kolegij;
			
	}
}

$pero = new Student("Petar", 1234);
// echo "<br>";
// $ivana = new Student("Ivana", 3431);
// echo "<br>";
// var_dump($pero);

// echo "<br>";
// $profa = new Profesor("Miljenko", array('PIN', 'DIMT', 'RIP'));
// echo "<br>";
// var_dump($profa);
// echo "<br>";
// $profa->ispisKolegija();